kerastuner/engine/tuner_utils.py

```diff
@@ -116,3 +117,28 @@ def average_histories(histories):
     # Convert {str: [float]} to [{str: float}]
     averaged = [dict(zip(metrics, vals)) for vals in zip(*averaged.values())]
     return averaged
+
+
+class UtilityFunction(object):
```

Comment:
Since this is specific to BayesianOptimization, this should live in ```kerastuner.tuners.bayesian.py```

I think we should make this into a class hierarchy, something like:

```python
class AcquisitionFunction(object):
  def __call__(self, x, gpr, y_max):
    raise NotImplementedError

class UCB(AcquisitionFunction):
  ...

class EI(AcquisitionFunction):
  ...
```

Reply:
This allows us to support users providing a subclass of this class to the BayesianOracle.

We also need to make sure we are sure what the signature of ```__call__``` should be